Description: Default to pwauth authenticator
 If no authenticator is specified, default to using /usr/sbin/pwauth
 in pipe mode.
 .
 libapache2-mod-authnz-external (3.2.6-1.netgear1) unstable; urgency=low
 .
   * Non-maintainer upload.
   * Hard-code pwauth authenticator.
Author: Justin Maggard <justin.maggard@netgear.com>

---

--- libapache2-mod-authnz-external-3.2.6.orig/mod_authnz_external.c
+++ libapache2-mod-authnz-external-3.2.6/mod_authnz_external.c
@@ -782,17 +782,10 @@ static authn_status authn_external_check
 	/* Get the path associated with that external */
 	if (!(extpath= apr_table_get(svr->auth_path, extname)))
 	{
-	    ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r,
-		"Invalid AuthExternal keyword (%s)", extname);
-	    return AUTH_GENERAL_ERROR;
+		extpath = "/usr/sbin/pwauth";
+		extmethod = "pipe";
 	}
-
-	/* Do the authentication, by the requested method */
-	extmethod= apr_table_get(svr->auth_method, extname);
-	if ( extmethod && !strcasecmp(extmethod, "function") )
-	    code= exec_hardcode(r, extpath, password);
-	else
-	    code= exec_external(extpath, extmethod, r, ENV_PASS, password);
+	code= exec_external(extpath, extmethod, r, ENV_PASS, password);
 
 	/* If return code was zero, authentication succeeded */
 	if (code == 0) return AUTH_GRANTED;
